### DISCUSSION TITLE
What skills/language/frameworks will help me get my first job the fastest?

### DISCUSSION THREAD

**Discussion by ArielMAJ** [Schweinepriester/github-profile-achievements#668] [Q&A] [answered]
Any suggestions?

I've always been interested in ML and Data Science, I have done some personal ML projects and I'm also studying it in my Master's degree, but it seems like there aren't many jobs for junior ML researchers out there. Should I keep specializing in ML or would studying something else help me land my first job faster? e.g. I'm also studying ReactJS/Node and also am somewhat proficient in C and some other languages/frameworks. I'm sure I can learn and become good at any language/framework, but there's too many to choose and I'm not sure what would be best. I mean, I need to pay my bills, I can try to get a job in the area I like after I have some kind of steady income/ability to do freelance.

**Comment by peterbaccaro** [comment, marked as answer]
As you are interested in ML and Data Science I would suggest that you learn Python since it is often used for ML and is the most popular language to learn if you look at the programming language indexes such as IEEE, PYPL and TIOBE.  It's also quite an easy language to learn.  After you have a decent understanding I would recommend that you learn about Object Oriented Programming and the SOLID principles and also learn about test driven development.

If you are interested in web development then React/Node is a very good choice.  If you are using Python then another option is to learn Shiny for Python to build the web frontend.

**Comment by bm4578** [comment]
If you're interested in web development, try the java back end or the react front end
